crates/ironrdp-rdpdr/src/pdu/efs.rs

```diff
@@ -0,0 +1,229 @@
+//! PDUs for [[MS-RDPEFS]: Remote Desktop Protocol: File System Virtual Channel Extension](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/34d9de58-b2b5-40b6-b970-f82d4603bdb5)
+
+use std::fmt::Debug;
+
+use ironrdp_pdu::utils::{encoded_str_len, write_string_to_cursor, CharacterSet};
+use ironrdp_pdu::{
+    cursor::{ReadCursor, WriteCursor},
+    PduEncode, PduResult,
+};
+use ironrdp_pdu::{invalid_message_err, unexpected_message_type_err, PduError};
+
+/// [2.2.1.1 Shared Header (RDPDR_HEADER)](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/29d4108f-8163-4a67-8271-e48c4b9c2a7c)
+/// A header that is shared by all RDPDR PDUs.
+#[derive(Debug)]
+pub struct SharedHeader {
+    pub component: Component,
+    pub packet_id: PacketId,
+}
+
+impl SharedHeader {
+    pub fn decode(src: &mut ReadCursor) -> PduResult<Self> {
+        Ok(Self {
+            component: src.read_u16().try_into()?,
+            packet_id: src.read_u16().try_into()?,
+        })
+    }
+
+    fn encode(&self, dst: &mut WriteCursor) -> PduResult<()> {
+        dst.write_u16(self.component as u16);
+        dst.write_u16(self.packet_id as u16);
+        Ok(())
+    }
+
+    fn size(&self) -> usize {
+        4 // 2 * u16  = 2 * 2 bytes = 4 bytes
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+#[repr(u16)]
+#[allow(non_camel_case_types)]
+pub enum Component {
+    RDPDR_CTYP_CORE = 0x4472,
```

Comment:
In `ironrdp` We usually transform enum variant to camel case for PDUs, And only for `bitflags!` do we preserve original case (see `ironrdp_cliprdr::pdu::FormatListResponse`  for example

Reply:
True! In the case of `bitflags!` generated structs, the reason is that those are actually `const` items.

Reply:
My preference is for overriding the standard Rust linter rules here because it makes it easier for me to cross reference from the codebase to the documentation. 

For example, if I Google "[RDPDR_CTYP_CORE](https://www.google.com/search?q=RDPDR_CTYP_CORE)" or "[RDPDR_CTYP_CORE = 0x4472](https://www.google.com/search?q=RDPDR_CTYP_CORE+%3D+0x4472)", I get the exactly relevant piece of documentation at the top, vs if I try "[RdpdrCtypCore](https://www.google.com/search?q=rdp+RdpdrCtypCore)" or "[RdpdrCtypCore = 0x4472](https://www.google.com/search?q=RdpdrCtypCore+%3D+0x4472)" or "[rdp [MS-RDPEFS] RdpdrCtypCore](https://www.google.com/search?q=rdp+%5BMS-RDPEFS%5D+RdpdrCtypCore)" etc., the documentation doesn't show up.

It's also helpful in the other direction, i.e. if you're looking at the documentation and say "I wonder where this is implemented?", you can just grep for `RDPDR_CTYP_CORE`. Of course it's not that difficult to translate into camel case, but that's only because I know what to look for -- if I'm an outside developer interested in exploring the codebase anew, I won't have that knowledge. 

So I would propose that we treat such constants like this here and moving forward (and ideally go back and change the CamelCase ones elsewhere, longer term). But if you prefer consistency with the rest of the codebase over this concern I'm not going to fight back on that.

Reply:
To be honest, I don’t feel too strongly about this either. I agree it’s actually useful for cross referencing with documentation. At some places I actually added comments with the actual name from the documentation for that exact same purpose. Eventually, we’ll want consistency: but I’m not against changing the casing, the motivation is good enough for me. What’s your opinion on this @pacmancoder?

Reply:
I am more inclined to keep casing consistent with other parts of IronRDP. Let's add doc comments with original constant names and rename enum variants to follow standard Rust conventions

Reply:
> I am more inclined to keep casing consistent with other parts of IronRDP.

Just noting that we would be going back and change the CamelCase ones of other similar enums for consistency. Are you suggesting we make this change for all enums at once, or that we stick to the Rust conventions forever for all the enums?

Reply:
> we stick to the Rust conventions forever for all the enums?

Yeah, I think sticking to rust conventions would be better in my opinion

Reply:
Got you!

Reply:
8e609e20be22f1243b406585d01172ff0065ba58